service/src/main/java/org/ehrbase/repository/EhrFolderRepository.java

```diff
@@ -132,101 +132,149 @@ private void store(
     @Transactional
     public void update(List<EhrFolderRecord> folderRecordList, UUID contributionId, UUID auditId) {
 
-        UUID ehrId = folderRecordList.get(0).getEhrId();
-        Result<EhrFolderRecord> old = getLatest(ehrId);
-
-        boolean isDeleted = false;
-        if (old.isEmpty()) {
-
-            Result<EhrFolderHistoryRecord> history = getLatestHistory(ehrId);
-
-            if (history.isNotEmpty() && BooleanUtils.isTrue(history.get(0).getSysDeleted())) {
-
-                old = history.into(EHR_FOLDER);
-                isDeleted = true;
+        EhrFolderRecord rootFolder = findRoot(folderRecordList);
+        UUID ehrId = rootFolder.getEhrId();
+        int ehrFoldersIdx = rootFolder.getEhrFoldersIdx();
+        Result<EhrFolderRecord> oldHead = getFolderHead(ehrId, ehrFoldersIdx);
+
+        boolean isDeleted;
+        int oldVersion;
+        OffsetDateTime now;
+        EhrFolderHistoryRecord delRecord;
+        UUID rootId;
+        if (oldHead.isEmpty()) {
+            Optional<EhrFolderHistoryRecord> history = getLatestHistoryRoot(ehrId, ehrFoldersIdx);
+
+            isDeleted = history.map(x -> x.getSysDeleted())
+                    .filter(deleted -> deleted)
```

Comment:
Function.identity

Reply:
I wanted to have it bit more explicity, but I could a swee add a comment.